### PR TITLE
feat: rate-limit unauthenticated public API routes

### PR DESCRIPTION
**Pull request by dan2k3k4**
Adds rate limiting to the unauthenticated public API routes (register, register status, regions, and the health callback).

<details><summary><h3>Greptile Summary</h3></summary>

This PR adds rate limiting to the four unauthenticated public API routes (`/register` POST, `/register/{uuid}` GET, `/regions` GET, and the instance health callback) using Laravel named rate limiters defined in `AppServiceProvider::boot()`. Trusted internal IPs (configured via `POLYDOCK_TRUSTED_IPS`) and callers presenting a valid health token bypass the limits entirely.

- Named limiters (`register` at 10/min, `public-read` at 60/min, `instance-health` at 120/min) replace what would have been a shared anonymous throttle counter, giving each route its own per-IP bucket.
- A new `trusted_ips` config key is parsed from a comma-separated env var; test suites add `tearDown` cache flushes to prevent counter bleed between tests.
- One test — `it_allows_trusted_ips_to_bypass_registration_status_throttle` — passes `REMOTE_ADDR` as a header argument to `getJson` rather than via `withServerVariables`, so the trusted-IP guard never fires and the test fails at request 61 with a 429.
</details>

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

The production rate-limiting logic is correct; a broken test will cause CI to fail on the trusted-IP bypass path for the registration-status route.

The core implementation — named rate limiters, trusted-IP bypass, token bypass for instance-health — is well-structured and the logic is sound. One test (it_allows_trusted_ips_to_bypass_registration_status_throttle) passes REMOTE_ADDR as a header to getJson instead of using withServerVariables, so $request->ip() returns 127.0.0.1, the trusted-IP check fails, and the test fails at request 61 with 429. The equivalent test for the POST register route was fixed correctly in this PR; this getJson variant was missed.

tests/Feature/Controllers/Api/RegisterControllerTest.php — the it_allows_trusted_ips_to_bypass_registration_status_throttle test uses the wrong approach for spoofing the client IP.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| app/Providers/AppServiceProvider.php | Adds three named rate limiters (register, public-read, instance-health) to boot(). Trusted-IP bypass and valid-token bypass for instance-health are correctly evaluated at request time via closures. Logic looks sound. |
| config/polydock.php | Adds trusted_ips config key parsed from POLYDOCK_TRUSTED_IPS env var via explode/trim/array_filter. Empty-string and missing-env edge cases are handled correctly. |
| routes/api.php | Applies named throttle middleware to all four public routes. Each route now uses its own isolated counter (register, public-read, instance-health). |
| tests/Feature/Api/InstanceHealthApiTest.php | Adds three new rate-limit tests. Trusted-IP test correctly uses withServerVariables. tearDown flushes the cache to prevent cross-test leakage. No issues. |
| tests/Feature/Controllers/Api/RegisterControllerTest.php | it_allows_trusted_ips_to_bypass_registration_status_throttle passes REMOTE_ADDR as a header argument to getJson instead of using withServerVariables, so the trusted-IP bypass never triggers and the test fails at request 61 with 429. |

</details>

<details><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[Incoming Request] --> B{throttle middleware}
    B --> C{Named limiter closure}
    C --> D{IP in trusted_ips?}
    D -->|Yes| E[Limit::none — bypass]
    D -->|No| F{instance-health route?}
    F -->|Yes| G{Valid health token?}
    G -->|Yes| E
    G -->|No| H[Limit::perMinute 120 by IP]
    F -->|No| I{register route?}
    I -->|Yes| J[Limit::perMinute 10 by IP]
    I -->|No| K[Limit::perMinute 60 by IP]
    H --> L{Under limit?}
    J --> L
    K --> L
    L -->|Yes| M[Proceed to controller]
    L -->|No| N[429 Too Many Requests]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[Incoming Request] --> B{throttle middleware}
    B --> C{Named limiter closure}
    C --> D{IP in trusted_ips?}
    D -->|Yes| E[Limit::none — bypass]
    D -->|No| F{instance-health route?}
    F -->|Yes| G{Valid health token?}
    G -->|Yes| E
    G -->|No| H[Limit::perMinute 120 by IP]
    F -->|No| I{register route?}
    I -->|Yes| J[Limit::perMinute 10 by IP]
    I -->|No| K[Limit::perMinute 60 by IP]
    H --> L{Under limit?}
    J --> L
    K --> L
    L -->|Yes| M[Proceed to controller]
    L -->|No| N[429 Too Many Requests]
```

</a>
</details>

<sub>Reviews (5): Last reviewed commit: ["feat: rate-limit unauthenticated public ..."](https://github.com/amazeeio/polydock-engine/commit/b022d68d54badf8fbfddd064e3a030476cdd103d) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=41060903)</sub>

### CODE REVIEW

File: tests/Feature/Controllers/Api/RegisterControllerTest.php [hunk 1]
```diff
@@ -87,4 +87,23 @@ public function it_redacts_sensitive_data_when_showing_registration_status(): vo
...
+
+        // WHEN we post the registration endpoint up to its per-minute limit (10)
+        for ($i = 0; $i < 10; $i++) {
+            $this->postJson('/api/register', $payload);
+        }
+
+        // THEN the 11th request within the same minute is rejected with 429
+        $this->postJson('/api/register', $payload)
+            ->assertStatus(429);
+    }
```
[the remark this answers is not in the document]
Comment (dan2k3k4) [reply tests/Feature/Controllers/Api/RegisterControllerTest.php:118]: Fixed — added a tearDown that flushes the cache so rate-limit counters reset between tests, making this order-independent.

File: routes/api.php [hunk 2]
```diff
@@ -30,12 +30,18 @@
...
-Route::get('/regions', [RegionsController::class, 'index'])->name('regions.index');
+Route::get('/regions', [RegionsController::class, 'index'])
+    ->name('regions.index')
+    ->middleware('throttle:60,1');
 
 Route::match(['get', 'post'], '/instance/{uuid}/health/{status}', [
     PolydockInstanceHealthController::class,
     '__invoke',
-])->name('api.instance.health');
+])->name('api.instance.health')->middleware('throttle:120,1');
```
[the remark this answers is not in the document]
Comment (dan2k3k4) [reply routes/api.php]: Fixed — switched to named rate limiters (register / public-read / instance-health) defined in AppServiceProvider, so each route has its own counter instead of the shared sha1(domain|ip) anonymous bucket.
